static/elements/chromedash-upcoming.js

```diff
@@ -52,12 +52,110 @@ class ChromedashUpcoming extends LitElement {
       loginUrl: {type: String},
       starredFeatures: {type: Object}, // will contain a set of starred features
       cardWidth: {type: Number},
+      lastFetchedOn: {type: Number},
+      lastMilestoneVisible: {type: Number},
+      milestoneArray: {type: Array},
+      milestoneInfo: {type: Object},
+      cardsToFetchInAdvance: {type: Number},
     };
   }
 
+  set lastFetchedOn(val) {
+    let oldVal = this._lastFetchedOn;
+    this._lastFetchedOn = val;
+    this.fetchNextBatch().then(()=>{
+      this.requestUpdate('lastFetchedOn', oldVal);
+    }).catch(() => {
+      alert('Some error occurred. Please refresh the page or try again later.');
+    });
+  }
+
+  get lastFetchedOn() {
+    return this._lastFetchedOn;
+  }
+
   constructor() {
     super();
     this.cardWidth = this.computeWidthofCard();
+    this.starredFeatures = new Set();
+    this.milestoneArray = [];
+    this.milestoneInfo = {};
+    this.cardsToFetchInAdvance = 3;
+  }
+
+  async fetchNextBatch() {
+    // promise to fetch next milestones
+    const nextMilestones = window.csClient.getSpecifiedChannels(this._lastFetchedOn+1+1,
+      this._lastFetchedOn+1+this.cardsToFetchInAdvance);
+
+    let tempMilestoneArray = [];
+    for (let i = 1; i <= this.cardsToFetchInAdvance; i++) {
+      tempMilestoneArray.push(this._lastFetchedOn+1+i);
+    }
+
+    // promise to fetch features in next milestones
+    let milestoneFeaturePromise = {};
+    tempMilestoneArray.forEach((milestone) => {
+      milestoneFeaturePromise[milestone] = window.csClient.getFeaturesInMilestone(milestone);
+    });
+
+    let tempMilestoneInfo;
+    let milestoneFeatures = {};
+
+    try {
+      tempMilestoneInfo = await nextMilestones;
```

Comment:
Try changing "tempMilestoneInfo" to "fetchedMilestoneInfo" to make it more descriptive.

Reply:
As you suggested in one of the previous comments, I replaced `tempMilestoneInfo` with `newMilestonesInfo` in [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1458/commits/d8659bbca508ca50367ee7ce97f2579a8d3f5379) commit. Is this okay ?